main/rest/leaf.py

```diff
@@ -229,24 +238,111 @@ def _get(self, params):
     def _patch(self, params):
         obj = Leaf.objects.get(pk=params['id'], deleted=False)
         model_dict = obj.model_dict
+        grandparent = obj.parent
 
         # Patch common attributes.
         if 'name' in params:
             obj.name = params['name']
-            obj.save()
+        
         new_attrs = validate_attributes(params, obj)
         obj = patch_attributes(new_attrs, obj)
+        obj.save()
+
+        # Change the parent of a leaf
+        if 'parent' in params:
+            if params['parent'] != params['id']:
+                children = Leaf.objects.filter(parent=obj.id, deleted=False)
+                if params['parent'] == None or params['parent'] == -1:
```

Comment:
This line isn't "wrong", but convention is to use `is` when comparing against `None`
```suggestion
                if params['parent'] is None or params['parent'] == -1:
```